### PR TITLE
Add dataDetectorTypes at textinput

### PR DESCRIPTION
**Pull request by Yummy-sk**
A PR that adds dataDetectorTypes to textInput.

[dataDetectorTypes](https://reactnative.dev/docs/textinput#datadetectortypes-ios)

### CODE REVIEW

**Comment by mununki** [comment]
I think binding to an array would be more scalable, what do you think? https://reactnative.dev/docs/textinput#datadetectortypes-ios:~:text=You%20can%20provide%20one%20type%20or%20an%20array%20of%20many%20types.

**Comment by Yummy-sk** [comment]
> I think binding to an array would be more scalable, what do you think? https://reactnative.dev/docs/textinput#datadetectortypes-ios:~:text=You%20can%20provide%20one%20type%20or%20an%20array%20of%20many%20types.

Thanks! When I look at the types, I get the following, so I think it would be better to handle it as an array as you said.

```
dataDetectorTypes?: DataDetectorTypes | DataDetectorTypes[] | undefined;
```

File: src/components/TextInput.res [hunk 1]
```diff
@@ -217,6 +217,8 @@ type textContentType = [
   | #oneTimeCode
 ]
 
+type dataDetectorTypes = [#phoneNumber | #link | #address | #calendarEvent | #none | #all]
+
```
Comment (MoOx) [inline src/components/TextInput.res:221]:
```suggestion
```

**Review by MoOx — CHANGES_REQUESTED** [review]
I guess we could do it this way ?

File: src/components/TextInput.res [hunk 2]
```diff
@@ -233,6 +233,7 @@ external make: (
   ~contextMenuHidden: bool=?,
   ~defaultValue: string=?,
   ~disableFullscreenUI: bool=?,
+  ~dataDetectorTypes: array<Text.dataDetectorTypes>=?,
```
Comment (MoOx) [inline src/components/TextInput.res:236]:
```suggestion
  ~dataDetectorTypes: array<Text.dataDetectorType>=?,
```

**Review by MoOx — APPROVED** [review]

**Comment by Yummy-sk** [comment]
@MoOx  Sorry for the late reply, I think the implementation you suggested is better :)

**Pull request merged**